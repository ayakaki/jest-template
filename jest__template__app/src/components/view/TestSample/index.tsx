import Link from 'next/link';

const TestSample = () => {
  return (
    <main>
      <h1>Nextjs+Jestのサンプルサプリ</h1>
      <p>設定がすごく楽になりました。</p>
      <p>スナップショット後に付け足されたよ</p>
      <Link href='jump-dst-direct' data-testid="jump-dst-direct-flg">
        <a>▶︎ 遷移先ページ（直接） へ遷移する</a>
      </Link>
    </main>
  );
};

export default TestSample;
