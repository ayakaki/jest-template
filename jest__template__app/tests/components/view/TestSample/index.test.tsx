import TestSample from "../../../../src/components/view/TestSample";
import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import "@testing-library/jest-dom/extend-expect";


describe("Sampleコンポーネント", () => {
  test("スナップショットテスト2", () => {
    const html = render(<TestSample />);
    expect(html).toMatchSnapshot();
  });


});