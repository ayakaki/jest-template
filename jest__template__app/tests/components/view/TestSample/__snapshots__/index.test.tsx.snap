// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sampleコンポーネント スナップショットテスト2 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <h1>
          Nextjs+Jestのサンプルサプリ
        </h1>
        <p>
          設定がすごく楽になりました。
        </p>
        <p>
          スナップショット後に付け足されたよ
        </p>
        <a
          href="/jump-dst-direct"
        >
          ▶︎ 遷移先ページ（直接） へ遷移する
        </a>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <h1>
        Nextjs+Jestのサンプルサプリ
      </h1>
      <p>
        設定がすごく楽になりました。
      </p>
      <p>
        スナップショット後に付け足されたよ
      </p>
      <a
        href="/jump-dst-direct"
      >
        ▶︎ 遷移先ページ（直接） へ遷移する
      </a>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
